src/preact/base-element.js

```diff
@@ -725,6 +728,26 @@ function collectProps(Ctor, element, ref, defaultProps) {
   return props;
 }
 
+/**
+ * Clones an Element into a VNode.
+ * @param {!Element} element
+ * @return {!PreactDef.Renderable}
+ */
+function createVNode(element) {
+  const {attributes} = element;
+  const props = {key: element};
+  for (let i = 0; i < attributes.length; i++) {
+    const {name, value} = attributes[i];
+    props[name] = value;
+  }
+  const children = [];
+  for (let i = 0; i < element.children.length; i++) {
+    // TODO(alanorozco): Could these be text nodes?
+    children.push(createVNode(element.children[i]));
+  }
+  return <element.tagName {...props}>{children}</element.tagName>;
```

Comment:
Is this actually allowed in JSX? I thought it's either has to be a constant string like "div" or a capital-case `Component`.

Alternatively, we can just call `Preact.createElement` directly. That would avoid double-allocation on the `...props`.

Reply:
It's allowed! You eventually just get a string in the call.

But changed to `createElement` to prevent double alloc.